.github/workflows/nx-cloud-agents.yml

```diff
@@ -48,44 +51,63 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name: Detect package manager
+        id: package_manager
+        run: |
+          echo "::set-output name=name::$([[ -f ./yarn.lock ]] && echo "yarn" || ([[ -f ./pnpm-lock.yaml ]] && echo "pnpm") || "npm")"
+
       # Set node/npm/yarn versions using volta, with optional overrides provided by the consumer
       - uses: volta-cli/action@fdf4cf319494429a105efaa71d0e5ec67f338c6e
         with:
           node-version: "${{ inputs.node-version }}"
           npm-version: "${{ inputs.npm-version }}"
           yarn-version: "${{ inputs.yarn-version }}"
 
+      # Install pnpm with exact version provided by consumer or fallback to latest
+      - name: Install PNPM
+        if: steps.package_manager.outputs.name == 'pnpm'
+        uses: pnpm/action-setup@v2.2.1
+        with:
+          version: ${{ inputs.pnpm-version || 'latest' }}
+
       - name: Print node/npm/yarn versions
         id: versions
         run: |
           node_ver=$( node --version )
           yarn_ver=$( yarn --version || true )
+          pnpm_ver=$( pnpm --version || true )
 
           echo "Node: ${node_ver:1}"
           echo "NPM: $( npm --version )"
           if [[ $yarn_ver != '' ]]; then echo "Yarn: $yarn_ver"; fi
+          if [[ $pnpm_ver != '' ]]; then echo "PNPM: $pnpm_ver"; fi
 
           echo "::set-output name=node_version::${node_ver:1}"
-          echo "::set-output name=use_yarn::$([[ -f ./yarn.lock ]] && echo "true" || echo "false")"
 
       - name: Use the node_modules cache if available [npm]
-        if: steps.versions.outputs.use_yarn != 'true'
+        if: steps.package_manager.outputs.name == 'npm'
         uses: actions/cache@v2
         with:
           path: ~/.npm
           key: ${{ runner.os }}-node-${{ steps.versions.outputs.node_version }}-${{ hashFiles('**/package-lock.json') }}
           restore-keys: |
             ${{ runner.os }}-node-${{ steps.versions.outputs.node_version }}-
 
+      - name: Use the node_modules cache if available [pnpm]
```

Comment:
This is not wiring up the cache, despite its name.

An actions/cache with pnpm example was submitted (although not yet merged for some reason) here: https://github.com/actions/cache/pull/585